Mathlib/Combinatorics/Quiver/ConnectedComponent.lean

```diff
@@ -57,11 +65,208 @@ end WeaklyConnectedComponent
 
 variable {V}
 
--- Without the explicit universe level in `Quiver.{v+1}` Lean comes up with
--- `Quiver.{max u_2 u_3 + 1}`. This causes problems elsewhere, so we write `Quiver.{v+1}`.
 /-- A wide subquiver `H` of `Symmetrify V` determines a wide subquiver of `V`, containing an
 arrow `e` if either `e` or its reversal is in `H`. -/
 def wideSubquiverSymmetrify (H : WideSubquiver (Symmetrify V)) : WideSubquiver V :=
   fun _ _ ↦ { e | H _ _ (Sum.inl e) ∨ H _ _ (Sum.inr e) }
 
+/-!
+## Strongly connected components (directed connectivity)
+We reuse `IsPreconnected V` for directed strong connectivity and add a positive-length variant.
+We then define the equivalence relation for strongly connected components and the quotient type.
+-/
+
+section StronglyConnected
+
+variable (V : Type*) [Quiver V]
+
+/-- Positive-length directed connectivity: every pair of vertices is connected by some
+path of length > 0. -/
+def IsPreconnectedPos : Prop :=
+  ∀ i j : V, ∃ p : Path i j, 0 < p.length
+
+/-- Unfold `IsPreconnectedPos` to an existential over paths. -/
+lemma IsPreconnectedPos_iff :
+    IsPreconnectedPos V ↔ ∀ i j : V, ∃ p : Path i j, 0 < p.length := Iff.rfl
+
+/-- From positive connectivity, get a positive-length path from `i` to `j`. -/
+lemma IsPreconnectedPos.exists
+    {V : Type*} [Quiver V] (h : IsPreconnectedPos V) (i j : V) :
+    ∃ p : Path i j, 0 < p.length :=
+  h i j
+
+/-- From positive connectivity, get a positive-length cycle at any vertex. -/
+lemma IsPreconnectedPos.exists_same
+    {V : Type*} [Quiver V] (h : IsPreconnectedPos V) (i : V) :
+    ∃ p : Path i i, 0 < p.length :=
+  h i i
+
+/-- Compatibility alias: ordinary strong connectivity (directed preconnectedness). -/
+abbrev IsStronglyConnected (V : Type*) [Quiver V] : Prop := IsPreconnected V
+
+/-- Compatibility alias: positive-length strong connectivity. -/
+abbrev IsSStronglyConnected (V : Type*) [Quiver V] : Prop := IsPreconnectedPos V
```

Comment:
I appreciate you are trying to reach a compromise between the old definition / terminology `IsPreconnected` and the new but we will be much better off if we have only one way to spell something. Even though an `abbrev` is quite a thin layer, it will break dot notation and generally just add to mental load.

I recommend:

- delete the existing `Quiver.IsPreconnected` (in the file `Combinatorics.Quiver.Symmetric`),
- not introduce `IsPreconnectedPos`
- make `IsStronglyConnected` and `IssStronglyConnected` into `def`s